home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/BaseContactService.java

```diff
@@ -25,50 +21,30 @@
 import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Service;
 
+
 @Service
 @RequiredArgsConstructor
-public class ContactServiceImpl implements ContactService {
-
-    private final ContactRepository contactRepository;
+public abstract class BaseContactService implements ContactService {
 
-    private final UserRepository userRepository;
+    protected final ContactRepository contactRepository;
 
-    private final ServiceMapper mapper;
+    protected final ServiceMapper mapper;
 
     @Transactional
     @Override
-    public ContactDto createContact(Long userId, ContactDto createContactDto) {
-        User user = getUserById(userId);
-        if (Boolean.TRUE.equals(createContactDto.getMain())) {
-            List<Contact> allByUserIdAndType = contactRepository
-                .findAllByUserIdAndType(userId, mapper.convert(createContactDto.getType(), ContactType.class));
-            allByUserIdAndType
-                .stream()
-                .filter(Contact::getMain)
-                .findAny()
-                .ifPresent(contact -> {
-                    throw new AlreadyExistHomeException("User with id "
-                        + userId + " already has main " + createContactDto.getType() + " contact");
-                });
-        }
-
-        Contact contact = mapper.convert(createContactDto, Contact.class);
-        contact.setUser(user);
+    public ContactDto createContact(Long parentEntityId, ContactDto createContactDto) {
+        var contact = mapper.convert(createContactDto, Contact.class);
+        checkAndFillParentEntity(createContactDto, contact, parentEntityId);
         contact.setEnabled(true);
         contactRepository.save(contact);
         return mapper.convert(contact, ContactDto.class);
     }
 
-    private User getUserById(Long id) {
-        return userRepository.findById(id)
-            .orElseThrow(() -> new NotFoundHomeException("User with id " + id + " wasn't found"));
-    }
+    protected abstract void checkAndFillParentEntity(ContactDto contactDto, Contact createContact, Long parentEntityId);
 
     @Override
-    public ContactDto updateContact(Long id, ContactDto updateContactDto) {
-        Contact contact = contactRepository.findById(id)
-            .filter(Contact::getEnabled)
-            .orElseThrow(() -> new NotFoundHomeException("User with id:" + id + " is not found"));
+    public ContactDto updateContact(Long parentEntityId, Long contactId, ContactDto updateContactDto) {
+        var contact = findingAndCheckingContactParentEntity(contactId, parentEntityId);
```

Comment:
checkAndGetParentEntity

Reply:
I think checkAndGetContactByParentId more better becouse we return Contact.